Bottom: a9f20cc25349fe2594aeae1014c86512beaff711
Top:    5fe7b77f89825f3fb2f887164f7bd1fbde8d04da
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2014-04-09 16:07:53 -0700

Refresh of 0001-lazy-init.patch

---

diff --git a/src/cma.c b/src/cma.c
index b34472e..12c8599 100644
--- a/src/cma.c
+++ b/src/cma.c
@@ -256,7 +256,7 @@ static int ucma_dev_init(struct cma_device *cma_dev)
 	if (ret) {
 		fprintf(stderr, PFX "Fatal: unable to query RDMA device\n");
 		ret = ERR(ret);
-		goto err3;
+		goto err2;
 	}
 
 	cma_dev->port_cnt = attr.phys_port_cnt;
@@ -267,7 +267,7 @@ static int ucma_dev_init(struct cma_device *cma_dev)
 	pthread_mutex_unlock(&mut);
 	return 0;
 
-err1:
+err2:
 	ibv_close_device(cma_dev->verbs);
 	cma_dev->verbs = NULL;
 err1:
@@ -308,7 +308,7 @@ int ucma_init(void)
 		goto err2;
 	}
 		
-	cma_dev_array = calloc(dev_cnt, sizeof *cma_dev);
+	cma_dev_array = calloc(dev_cnt, sizeof *cma_dev_array);
 	if (!cma_dev_array) {
 		ret = ERR(ENOMEM);
 		goto err2;
@@ -344,8 +344,8 @@ struct ibv_context **rdma_get_devices(int *num_devices)
 		goto err1;
 
 	for (i = 0; i < cma_dev_cnt; i++) {
-		ucma_dev_init(cma_dev_array[i]);
-		devs[i] = cma_dev_array[i].verbs
+		ucma_dev_init(&cma_dev_array[i]);
+		devs[i] = cma_dev_array[i].verbs;
 		if (!devs[i])
 			goto err2;
 	}
